pbpy/pbtools.py

```diff
@@ -140,94 +154,144 @@ def remove_file(file_path):
+        handle_success()
+    elif "successfully rebased and updated" in out:
+        handle_success()
+    elif "failed to merge in the changes" in out or "could not apply" in out:
+        handle_error("Aborting the rebase. Changes on one of your commits will be overridden by incoming changes. Please request help in #tech-support to resolve conflicts, and please do not run StartProject.bat until the issue is resolved.")
+    elif "unmerged files" in out or "merge_head exists" in out:
+        # we can't abort anything, but don't let stash linger to restore the original repo state
+        pop_if_stashed()
+        error_state("You are in the middle of a merge. Please request help in #tech-support to resolve it, and please do not run StartProject.bat until the issue is resolved.", fatal_error=True)
+    elif "unborn" in out:
+        if should_attempt_auto_resolve():
+            pblog.error("Unborn branch detected. Retrying...")
+            retry_count += 1
+            resolve_conflicts_and_pull(retry_count)
+        else:
+            handle_error("You are on an unborn branch. Please request help in #tech-support to resolve it, and please do not run StartProject.bat until the issue is resolved.")
+    elif "no remote" in out or "no such remote" in out or "refspecs without repo" in out:
+        if should_attempt_auto_resolve():
+            pblog.error("Remote repository not found. Retrying...")
+            retry_count += 1
+            resolve_conflicts_and_pull(retry_count, 2)
+        else:
+            handle_error("The remote repository could not be found. Please request help in #tech-support to resolve it, and please do not run StartProject.bat until the issue is resolved.")
+    elif "cannot open" in out:
+        if should_attempt_auto_resolve():
+            pblog.error("Git file info could not be read. Retrying...")
+            retry_count += 1
+            resolve_conflicts_and_pull(retry_count, 3)
+        else:
+            handle_error("Git file info could not be read. Please request help in #tech-support to resolve it, and please do not run StartProject.bat until the issue is resolved.")
     else:
-        pblog.error("Aborting the rebase, an unknown error occured. Request help on #tech-support to resolve conflicts, and please do not run StartProject.bat until issue is solved.")
-        pbgit.abort_rebase()
-        pbgit.stash_pop()
-        error_state(True)
+        # We have no idea what the state of the repo is. Do nothing except bail.
+        error_state("Aborting the repo update because of an unknown error. Request help in #tech-support to resolve it, and please do not run StartProject.bat until the issue is resolved.", fatal_error=True)
+
+    if os.name == "nt":
+        subprocess.Popen("git lfs prune -c ; git lfs dedup", shell=True, creationflags=subprocess.DETACHED_PROCESS, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    elif os.name == "posix":
```

Comment:
We don't have linux support on PBSync, that can be removed.

Reply:
I don't think it's harming anything, and this is for Mac as well as Linux, if we can do anything to get us closer to supporting them, it will be better for our users.

Reply:
If that's the case, please implement that code piece in a seperate branch for linux support, and we can merge it when the whole source supports linux.

No linux support is better than halfly done linux support, because it makes the code more and more complex.

Reply:
I worry that branching the code will leave it out to rot. Mainlining this small bit of code will help us maintain it better.

Reply:
I don't agree, this code piece should go into separate branch.

Reply:
Will be discussed later, resolved.